### PR TITLE
Add obstacle debug draw

### PR DESCRIPTION
**Pull request by necessarily-equal**
This is a cleanup version of the code I used to debug bots' obstacles. I found it to be very useful.

Screenshots:
Screenshot from 2023-01-08 12-33-23
Screenshot from 2023-01-10 10-06-34
Screenshot from 2022-11-06 14-44-51

### CODE REVIEW

File: src/sgame/botlib/bot_nav_edit.cpp [hunk 1]
```diff
@@ -160,6 +219,14 @@ void BotDebugDrawMesh()
 		duDebugDrawNavMeshPortals( &dd, *cmd.nav->mesh );
 	}
 
+	//if ( cmd.showobstacles )
```
Comment (necessarily-equal) [inline src/sgame/botlib/bot_nav_edit.cpp:222]: I'm not sure if there is any point making this optional. I think it's fine as is.

**Comment by necessarily-equal** [comment]
To test, you need to use e.g. `/navedit enable level0` after a /devmap.

**Comment by slipher** [comment]
Why does it draw a different shape in the first screenshot than the other two?

**Comment by necessarily-equal** [comment]
That'd be https://github.com/Unvanquished/Unvanquished/issues/2337, there is a bug in the renderer that makes some lines disappear or that makes some lines between wrong vertex. Sometime you need a bit of patience to be sure. This is also why this is drawing so many lines, to have more chances some will be drawn correctly 😅

File: src/sgame/botlib/bot_nav_edit.cpp (hunk 1 above)
Comment (DolceTriade) [reply src/sgame/botlib/bot_nav_edit.cpp:222]: Unless there is a severe performance impact, I think it's fine to make it on by default. If we find a performance impact, we can add the check back.

**Comment by necessarily-equal** [comment]
Should I go ahead and merge this?

**Comment by DolceTriade** [comment]
lgtm

**Pull request merged**